src/SFML/Graphics/ImageLoader.cpp

```diff
@@ -227,35 +227,34 @@ bool ImageLoader::saveImageToFile(const std::string& filename, const std::vector
     if (!pixels.empty() && (size.x > 0) && (size.y > 0))
     {
         // Deduce the image type from its extension
-        if (filename.size() > 3)
+
+        // Extract the extension
+        const std::size_t dot = filename.find_last_of('.');
+        const std::string extension = dot != std::string::npos ? toLower(filename.substr(dot + 1)) : "";
```

Comment:
I thought about something similar as well, but I don't think it's worth making the code more elaborate. Also your snippet is excluding the case where there actually is a dot, but no extension (the file name could be `image.`). So you could actually check for an empty extension first to skip all further checks, but at the same time that's marginal complexity compared to everything else happening in that whole call.